### ISSUE TITLE
`no-unsafe-any` with tsx throwing error

### ISSUE THREAD

**Issue by tyv** [palantir/tslint#3080]
Hi! Can't understand why. 

### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: CLI 

#### TypeScript code being linted

```tsx
import * as React from 'react';

export class App extends React.Component<{}, {}> {
    public render(): JSX.Element {
        return (<span>asd</span>) as JSX.Element;
    }
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-unsafe-any": true
    }
}
```

#### Actual behavior
Error

```bash
ERROR: app.tsx[5, 17]: Unsafe use of expression of type 'any'.
ERROR: app.tsx[5, 28]: Unsafe use of expression of type 'any'.
```

#### Expected behavior
No errors

**Comment by adidahiya** [comment]
Do you have `@types/react` installed? Are there any `tsc` errors when you compile your project with `--noImplicitAny`?

**Comment by tyv** [comment]
@adidahiya 
> Do you have `@types/react` installed

yes

> Are there any `tsc` errors when you compile your project with `--noImplicitAny`?

no

**Comment by Nitive** [comment]
I have the same problem

**Comment by adidahiya** [comment]
Can you try to repro this with the latest TSLint? We've had various refactorings occur since 5.5.0.

**Comment by Ailrun** [comment]
It's still reproducible (in my case)

**Comment by bahkostya** [comment]
The same problem

**Comment by JoePotentier** [comment]
For anyone that finds this in the future. I was struggling with this for over an hour while configuring eslint. After adding the types with `yarn add -D @types/react @types/react-dom` I was still getting an error on the return statement. This was fixed by simply reloading/restarting vscode. Save yourself some time and use F1 -> `Reload Window`.

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
